src/Ui/Table.elm

```diff
@@ -0,0 +1,464 @@
+    , columns = Dict.empty
+    , extendableView = Nothing
+    , liftMsg = liftMsg
+    }
+
+
+addColumn :
+    ( String, Int -> ColumnConfig record msg )
+    -> TableConfig record msg
+    -> TableConfig record msg
+addColumn ( k, v ) config =
+    { config | columns = Dict.insert k (v <| Dict.size config.columns) config.columns }
+
+
+addExtendableView : (record -> Html msg) -> TableConfig record msg -> TableConfig record msg
+addExtendableView extendableView config =
+    { config | extendableView = Just extendableView }
+
+
+column :
+    String
+    -> Css.LengthOrAuto Css.Px
+    -> (record -> ColumnContent msg)
+    -> ColumnAlignment
+    -> Int
+    -> ColumnConfig record msg
+column name length content a =
+    ColumnConfig name length (always content) True a
+
+
+columnWithDataEntryId :
+    String
+    -> Css.LengthOrAuto Css.Px
+    -> (DataEntryId -> record -> ColumnContent msg)
+    -> ColumnAlignment
+    -> Int
+    -> ColumnConfig record msg
+columnWithDataEntryId name length content a =
+    ColumnConfig name length content True a
+
+
+hideColumn :
+    String
+    -> TableConfig record msg
+    -> TableConfig record msg
+hideColumn k =
+    setColumnVisbility k False
+
+
+showColumn :
+    String
+    -> TableConfig record msg
+    -> TableConfig record msg
+showColumn k =
+    setColumnVisbility k True
+
+
+setColumnVisbility :
+    String
+    -> Bool
+    -> TableConfig record msg
+    -> TableConfig record msg
+setColumnVisbility k visibility =
+    updateColumn k (\v -> { v | visible = visibility })
+
+
+setColumnWidth :
+    String
+    -> Css.LengthOrAuto Css.Px
+    -> TableConfig record msg
+    -> TableConfig record msg
+setColumnWidth k w =
+    updateColumn k (\v -> { v | width = w })
+
+
+setColumnAlignment :
+    String
+    -> ColumnAlignment
+    -> TableConfig record msg
+    -> TableConfig record msg
+setColumnAlignment k a =
+    updateColumn k (\v -> { v | alignment = a })
+
+
+updateColumn :
+    String
+    -> (ColumnConfig record msg -> ColumnConfig record msg)
+    -> TableConfig record msg
+    -> TableConfig record msg
+updateColumn k f config =
+    { config | columns = Dict.update k (Maybe.map f) config.columns }
+
+
+cellContentStyle : Attribute msg
+cellContentStyle =
+    css [ Css.flexGrow (Css.num 1) ]
+
+
+unwrapColumnContent : ColumnContent msg -> Html msg
+unwrapColumnContent (ColumnContent x) =
+    x
+
+
+cellColumns : Flex -> List (ColumnContent msg) -> ColumnContent msg
+cellColumns w =
+    ColumnContent
+        << div
+            [ css
+                ([ Css.displayFlex
+                 , Css.flexDirection Css.row
+                 , Css.height (Css.pct 100)
+                 , Css.whiteSpace Css.normal
+                 ]
+                    ++ flexToStyle w
+                )
+            ]
+        << List.map unwrapColumnContent
+
+
+
+-- Table.cellColumn Table.Grow [ row Table.Grow [ text ..., int... ], icon ..., ]
+
+
+cellText : TextStyle -> String -> ColumnContent msg
```

Comment:
Here it looks like the table defines the styling of the column contents. This seems a bit too much scope of responsibility. Arrnaging vs looks of each piece. Do we want this?

Reply:
We have an opt out, cellCustom. 

As I see it we want to make it as easy as possible to create tables that are consistent in the look and feel. I believe it can help by giving them ways to create text/numbers/links/ect. I'm a bit scared that we are going to have different fontsizes ect if we do not supply these helper functions.